diff --git a/src/MSSMNoFV_onshell.cpp b/src/MSSMNoFV_onshell.cpp
index 7ba57dd..b106ab4 100644
--- a/src/MSSMNoFV_onshell.cpp
+++ b/src/MSSMNoFV_onshell.cpp
@@ -198,6 +198,14 @@ void MSSMNoFV_onshell::check_problems() const
       if (!do_force_output())
          throw EPhysicalProblem(get_problems().get_problems());
    }
+   if (get_mu2().diagonal().minCoeff() < 0. ||
+       get_md2().diagonal().minCoeff() < 0. ||
+       get_mq2().diagonal().minCoeff() < 0. ||
+       get_me2().diagonal().minCoeff() < 0. ||
+       get_ml2().diagonal().minCoeff() < 0.) {
+      if (!do_force_output())
+         throw EPhysicalProblem("soft mass squared < 0");
+   }
 }
 
 void MSSMNoFV_onshell::copy_susy_masses_to_pole()
